test/libsolidity/SemanticTest.cpp

```diff
@@ -88,9 +88,15 @@ SemanticTest::SemanticTest(string const& _filename, langutil::EVMVersion _evmVer
 	if (m_runWithEwasm && !m_supportsEwasm)
 		m_runWithEwasm = false;
 
-	m_runWithABIEncoderV1Only = m_reader.boolSetting("ABIEncoderV1Only", false);
-	if (m_runWithABIEncoderV1Only && !solidity::test::CommonOptions::get().useABIEncoderV1)
-		m_shouldRun = false;
+	if (!solidity::test::CommonOptions::get().useABIEncoderV1)
+	{
+		for (auto const& [filename,source]: m_reader.sources().sources)
+			if (source.find("pragma abicoder v1;") != string::npos)
+			{
+				m_shouldRun = false;
+				break;
+			}
+	}
```

Comment:
You could use ranges-v3 here now. Something like this should do the trick:
```suggestion
	m_shouldRun = m_shouldRun && !(
		!solidity::test::CommonOptions::get().useABIEncoderV1 &&
		ranges::any_of(m_reader.sources().sources | views::values, [](string const& source) {
			source.find("pragma abicoder v1;") != string::npos
		})
	);
```

EDIT: But see my comment below first. Maybe this `if` is not necessary at all.